### PR TITLE
Migrate from spring-kafka to native Apache Kafka client

### PR DESCRIPTION
**Pull request by ctamisier**
PR candidate for fixing https://github.com/jhipster/generator-jhipster/issues/9287

### CODE REVIEW

**Comment by ctamisier** [comment]
Thanks for the review !
I've pushed some changes and I need to include the Gradle part as well.

**Comment by pascalgrimaud** [comment]
@ctamisier : don't hesitate to ping me as soon as you update this PR, so I can have a look again. If you don't have time, just tell me, I'll help you to update this, as I'd like to have this for the next release and start contribution on the entity part + Kafka

**Comment by ctamisier** [comment]
Since your last review I applied the following changes:

- Add Gradle support (build.gradle.ejs)
- Add `<logger name="org.apache.kafka" level="INFO"/>` (org.apache is in WARN) (To be discussed eventually)
- Remove `<logger name="kafka" level="WARN"/>` and `<logger name="org.I0Itec" level="WARN"/>`

I'm not really good about docker compose file definition... I tried to make `docker-compose -f src/main/docker/app.yml up` working but the app can't connect to kafka.

I don't know what `generator-jhipster/test/templates/compose/09-kafka/src/main/docker/kafka.yml` is for, so I didn't modify it. Should it be a "generation" of `generator-jhipster/generators/server/templates/src/main/docker/kafka.yml.ejs` ? (same with app.yml and app.yml.ejs)

We still need to adjust app.yml.ejs (I've an issue now because I use `depends_on` in kafka.yml.ejs).

Your help @pascalgrimaud is really welcome, I'm kind of learning docker-compose right now and I might not have the "Best Practices" to apply on this part.

**Comment by pascalgrimaud** [comment]
ok @ctamisier
I'll start the work on this PR tonight or tomorrow and will probably commit directly to your branch

**Comment by pascalgrimaud** [comment]
Just done the review and the tests.
Everything works fine ! So going to merge this as soon as the CI is green.

good job @ctamisier and @clement26695 !

@ctamisier : plz, don't rebase / force push, otherwise, we will lost our work here :)

File: generators/server/templates/pom.xml.ejs [hunk 1]
```diff
@@ -628,14 +628,18 @@
...
-            <artifactId>spring-kafka</artifactId>
-		</dependency>
+            <groupId>org.apache.kafka</groupId>
+            <artifactId>kafka-clients</artifactId>
+        </dependency>
         <dependency>
-            <groupId>org.springframework.kafka</groupId>
-            <artifactId>spring-kafka-test</artifactId>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>database-commons</artifactId>
```
Comment (bsideup) [inline generators/server/templates/pom.xml.ejs]: FYI this one is not needed for Kafka I guess

Comment (ctamisier) [reply generators/server/templates/pom.xml.ejs]: You're right! thanks.
https://github.com/jhipster/generator-jhipster/pull/10598

**Pull request merged**